carl/context/context_space.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+from typing import Any, List, Union
+
+import gymnasium.spaces as spaces
+import numpy as np
+from ConfigSpace.configuration_space import Configuration, ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    Hyperparameter,
+    NormalFloatHyperparameter,
+    NumericalHyperparameter,
+    UniformFloatHyperparameter,
+)
+from omegaconf import DictConfig
+from carl.context.search_space_encoding import search_space_to_config_space
+from typing_extensions import TypeAlias
+from carl.utils.types import Context, Contexts
+
+ContextFeature: TypeAlias = Hyperparameter
+NumericalContextFeature: TypeAlias = NumericalHyperparameter
+NormalFloatContextFeature: TypeAlias = NormalFloatHyperparameter
+UniformFloatContextFeature: TypeAlias = UniformFloatHyperparameter
+
+
+class ContextSpace(object):
+    def __init__(
+            self,
+            context_features: list[ContextFeature]
```

Comment:
Why do we use a list here? I don't see any advantage over a dict, especially given that we turn it into a dict anyway

Reply:
I am fine with both. Dict would be more verbose

Reply:
Since this is the feature of CARL people are likely least familiar with, I think verbose is better.

Reply:
updated